### DISCUSSION TITLE
MongoDB query rules: $and should not be default on root query level

### DISCUSSION THREAD

**Discussion by WebDevMichael** [react-querybuilder/react-querybuilder#339] [Ideas]
Hi,

I'm working on simplifying the mongodb query string output. At the moment all MongDB queries are surrounded with '$and' at the root of the query.

For example, this is the current output for a simple rule: `{"$and":[{"firstName":{$eq:"Test"}}]}`
This can be simplified to: `{"firstName":"Test"}`

**Feature:** When the root combinator is '$and', it should not be applied in the query. Also as you can see, the '$eq' is also redundant.

I already made many changes for this and am changing and expanding the test files.

Thanks again for this project

**Comment by WebDevMichael** [comment]
But then again, '$and' should be there at the root when there is at least 1 '$or' for example 🤔

**Comment by WebDevMichael** [comment]
Never mind my last comment, this is a valid query, no $and needed:
`{"firstName":"Test","$or":[{"lastName":"Test2"},{"numField":{"$in":["1","2"]}}]}`

**Comment by jakeboone02** [comment]
Hi @WebDevMichael. Thanks for the suggestions, they look like good improvements! I do like the simpler format. (I had never actually used MongoDB in a real environment when we added the export format, I just verified the logic by checking the documentation.)

Do you intend to submit a PR with the modifications? I ask because it looks like you've already published a [fork of `react-querybuilder`](https://www.npmjs.com/package/@frontlinetech/react-querybuilder), and [your GitHub fork of this repo](https://github.com/WebDevMichael/react-querybuilder) has some changes on its `main` and [`5.4.3`](https://github.com/WebDevMichael/react-querybuilder/tree/5.4.3) branches that would prevent an upstream merge into this repo.

You're _absolutely_ free to fork this project and publish an alternative version of course, but we wouldn't be able to support a fork directly. Let me know what you want to do.
